### ISSUE TITLE
[Q] How can I paint something as an inversion of color on what has been drawn before?

### ISSUE THREAD

**Issue by NightMachinary** [JuliaGraphics/Luxor.jl#110]
I want to show some debug info on a simulation's animation. I have tried this thus far, but it needs very big fontsizes and is unsuitable:
![image](https://user-images.githubusercontent.com/36224762/90340329-9f982880-e00c-11ea-8e56-f2306052aad1.png)
I want to be able to draw those "#X" numbers as a color-inversion mask on what has been drawn underneath it. How can I do that?

**Comment by cormullion** [comment]
Hi! Not too sure what you're looking for, but you could experiment with the compositing modes.

```
using Luxor

@draw begin
   img = readpng("/tmp/testimage.png")
   placeimage(img, O, centered=true)   
    setmode("dest_atop")
    sethue("black")
    fontsize(80)
    translate(-150, -150)
    box(O, 100, 100, :clip)
    text("#42", O, halign=:center)
    clipreset()
end 
```
<img width="542" alt="Screenshot 2020-08-16 at 18 49 25" src="https://user-images.githubusercontent.com/52289/90340592-37881900-dff1-11ea-9dcf-7d65e5ce05c4.png">

They're tricky little blighters, but you may be able to get something out. I seem to recall that clipping regions were required. The Cairo documentation (if you can find it) has more information about them...

**Comment by NightMachinary** [comment]
I solved my initial problem with a better design choice:
![image](https://user-images.githubusercontent.com/36224762/90341220-08829f00-e013-11ea-8299-54e4ed493de7.png)

But I still would love to know how to paint an inversion mask using Luxor. I did try the blend modes, but they did not have an inversion mode, and the ones they did have just turned my text creamy ghostish white. I guess I'll close the issue, as the Cairo docs seem to be the way to go. I thought there might be an easy solution.
